opm/grid/cpgrid/CpGridUtilities.cpp

```diff
@@ -65,4 +69,100 @@ lgrIJK(const Dune::CpGrid& grid, const std::string& lgr_name)
     return std::make_tuple(cellIdxToLgrCartesianIdx, lgrCartesianIdxToCellIdx, lgrIJK);
 }
 
+std::vector<std::array<double, 6>> lgrCOORD(const Dune::CpGrid& grid,
+                                            int level,
+                                            const std::unordered_map<int, int>&  lgrCartesianIdxToCellIdx,
+                                            const std::vector<std::array<int, 3>>& lgrIJK)
+{
+    // LGR dimensions
+    const auto& lgr_dim = grid.currentData()[level]->logicalCartesianSize();
+    const int nx = lgr_dim[0];
+    const int ny = lgr_dim[1];
+
+
+    // Initialize all pillars as inactive (setting COORD values to std::numeric_limits<double>::max()).
+    std::vector<std::array<double,6>> lgrCOORD((nx+1)*(ny+1));
+    for (auto& pillar : lgrCOORD) {
+        pillar.fill(std::numeric_limits<double>::max());
+    }
+
+    // Map to all k values (per pillar) grouped by (i,j)
+    std::map<std::pair<int, int>, std::vector<int>> pillars;
+
+    // Group elements by first two entries (i,j)
+    for (const auto& ijk : lgrIJK) {
+        pillars[std::make_pair(ijk[0], ijk[1])].push_back(ijk[2]);
+    }
+
+    // Rewrite values for active pillars
+    for (int j = 0; j < ny; ++j) {
+        for (int i = 0; i < nx; ++i) {
+            auto it = pillars.find(std::make_pair(i,j));
+            if (it == pillars.end()) {
+                continue; // no active pillar at (i,j)
+            }
+
+            const auto& pillar = it->second; // vector with all k's attached to (i,j)
+
+            // Get min/max k for pillar at (i,j)
+            auto [bottom_k, top_k] = std::minmax_element(pillar.begin(), pillar.end());
+
+            const auto bottom_lgr_cartesian_idx = ((*bottom_k)*nx*ny) + (j*nx) + i;
+            const auto top_lgr_cartesian_idx = ((*top_k)*nx*ny) + (j*nx) + i;
+
+            const auto& bottomElemIdx = lgrCartesianIdxToCellIdx.at(bottom_lgr_cartesian_idx);
+            const auto& topElemIdx = lgrCartesianIdxToCellIdx.at(top_lgr_cartesian_idx);
+
+            const auto& levelGrid = *(grid.currentData()[level]);
+
+            const auto& bottomElem = Dune::cpgrid::Entity<0>(levelGrid, bottomElemIdx, true);
+            const auto& topElem = Dune::cpgrid::Entity<0>(levelGrid, topElemIdx, true);
+
+            // Recall that a cell has 8 corners:
+            //        6 --- 7
+            //       /     /   TOP FACE
+            //      4 --- 5
+            //        2 --- 3
+            //       /     /   BOTTOM FACE
+            //      0 --- 1
```

Comment:
Maybe move this comment to the next function?

Reply:
Moved!